### ISSUE TITLE
tf.nn.softmax can give a result, when input shape is [2,3] and axis=2, Is it a bug?

### ISSUE THREAD

**Issue by guotong1988** [tensorflow/tensorflow#22793]
### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**:
windows 10
- **Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device**:
- **TensorFlow installed from (source or binary)**:
binary
- **TensorFlow version (use command below)**:
1.10.0
- **Python version**:
3.5.2
- **Bazel version (if compiling from source)**:
- **GCC/Compiler version (if compiling from source)**:
- **CUDA/cuDNN version**:
no CUDA
- **GPU model and memory**:
no GPU
- **Exact command to reproduce**:

### Describe the problem

for  `tf.nn.softmax` , the `axis=2` but the input tensor shape is `[2,3]`.
And the result is 

```
tf.Tensor(
[[0.5 0.5 0.5]
 [0.5 0.5 0.5]], shape=(2, 3), dtype=float32)
```

I think the axis should `>=0` and `<=1`

if the `axis>=2`, it should be error.

### Source code / logs

```
import tensorflow as tf
tf.enable_eager_execution()
ones = tf.ones(shape=[2,3])
temp = tf.nn.softmax(ones,axis=2)
print(temp)
```

**Comment by yongtang** [comment]
Added a PR #22849 for the fix.

**Comment by tensorflowbutler** [comment]
Nagging Assignees @drpngx, @harshini-gadige: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.
